Pilot/pilotCommands.py

```diff
@@ -298,7 +301,24 @@ def _locateInstallationScript(self):
         except OSError:
             pass
 
-    def _installDIRAC(self):
+    def _getPreinstalledEnvScript(self):
+        """ Get preinstalled environment script if any """
+
+        preinstalledEnvScript = self.pp.preinstalledEnv
+        if not preinstalledEnvScript and self.pp.preinstalledEnvPrefix:
+            version = self.pp.releaseVersion or "pro"
+            arch = platform.system() + "-" + platform.machine()
+            preinstalledEnvScript = os.path.join(self.pp.preinstalledEnvPrefix, version, arch, "diracosrc")
+
+        self.log.info("Evaluating env script %s" % preinstalledEnvScript)
+        if not safe_listdir(preinstalledEnvScript):
```

Comment:
This is calling `safe_listdir` on a file. Which results in "Not a directory"
```
touch aFile; python -c "import os; os.listdir('aFile')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
OSError: [Errno 20] Not a directory: 'aFile'
```

Reply:
PS: with python 3.9
```
touch aFile; python -c "import os; os.listdir('aFile')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NotADirectoryError: [Errno 20] Not a directory: 'aFile'
```

Reply:
Ouf, thanks. Corrected.